### PR TITLE
Cookiecutter

### LINKED ISSUE #1: Generate boilerplate with cookiecutter-pypackage
I'll be opening a Pull Request to add some boilerplate based on https://github.com/Ouranosinc/cookiecutter-pypackage that will provide some structure to the project.

Please don't add anything to the `main` branch until then, thanks!

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
  - This PR fixes #1 
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [ ] CHANGELOG.rst has been updated (with summary of main changes).
  - [ ] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Finalizes the cookiecutter template needed to begin development

### Does this PR introduce a breaking change?

No.

### Other information:

Refer to https://github.com/Ouranosinc/cookiecutter-pypackage

Numpydoc was disabled due to issue with their CLI on conda and PyPI (pre-commit hooks work as intended).

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@coxipi This is ready for final review and merge.

I'll tag a v0.1.0 release once this PR is merged.

File: .github/workflows/bump-version.yml
Comment (coxipi) [inline .github/workflows/bump-version.yml:36]: I'm confused about the change, since it's a whole new repo. Is this something about CookieCutter previously not respecting this structure with `src/...`?

**Review by coxipi — APPROVED** [review]
Looks good to me, although I have very low knowledge on this.

File: .github/workflows/bump-version.yml
Comment (coxipi) [reply .github/workflows/bump-version.yml:36]: The first commit was a generic Github template, and now you're applying your own CookieCutter template?

Comment (Zeitsperre) [reply .github/workflows/bump-version.yml:36]: There are a lot of moving parts and sometimes I forget to add certain files. The `src` structure is relatively, and I'm still moving the template over to it, so it seems.

I'll probably be pushing one more change (the issue about numpydoc that I raised is fixed). I'll merge soon after!

**Pull request merged**
